pkpdapp/pkpdapp/views/dataset.py

```diff
@@ -0,0 +1,200 @@
+    fields = ['name', 'description']
+    template_name = 'dataset_form.html'
+
+
+def create(request):
+    context = {}
+    if request.method == "POST":
+        form = CreateNewDataset(request.POST, request.FILES)
+        context["form"] = form
+
+        if form.is_valid():
+            uploaded_file = request.FILES['file']
+
+            # handling errors in uploaded files
+
+            # error in file type
+            if not uploaded_file.name.endswith('.csv'):
+                messages.error(request,
+                               BASE_FILE_UPLOAD_ERROR +
+                               'THIS IS NOT A CSV FILE.')
+                return render(request, 'dataset_create.html',
+                              context)
+
+            # error in file content
+            data = pd.read_csv(uploaded_file)
+            colnames = list(data.columns)
+            if len(colnames) > 4:
+                messages.error(
+                    request,
+                    BASE_FILE_UPLOAD_ERROR +
+                    'THIS FILE HAS TOO MANY COLUMNS. ' +
+                    'IT SHOULD ONLY HAVE: subject id, time, biomarker type, ' +
+                    'value')
+                return render(request, 'dataset_create.html',
+                              context)
+            required_cols = ['subject id', 'time', 'biomarker type', 'value']
+            error_cols = []
+            error_string = (BASE_FILE_UPLOAD_ERROR +
+                            'FILE DOES NOT CONTAIN: ')
+            for col in required_cols:
+                if col not in colnames:
+                    error_cols.append(col)
+            if len(error_cols) > 0:
+                messages.error(request,
+                               error_string + ', '.join(error_cols))
+                return render(request, 'dataset_create.html',
+                              context)
+
+            # no errors -> process and save as dataset
+            dataset = Dataset(
+                name=form.cleaned_data["name"],
+                description=form.cleaned_data["description"],
+                datetime=form.cleaned_data["datetime"],
+                administration_type=form.cleaned_data["administration_type"],
+            )
+            dataset.save()
+
+            # add to demo project
+            Project = apps.get_model("pkpdapp", "Project")
+            demo_project = Project.objects.get(name='demo')
+            demo_project.datasets.add(dataset)
+            context["dataset"] = dataset
+
+            # find all the biomarker types for that dataset
+            bts_unique = data["biomarker type"].unique().tolist()
+            request.session['bts_unique'] = bts_unique
+            request.session['data_raw'] = data.to_json()
+            return redirect('dataset-biomarkers')
+
+    else:
+        form = CreateNewDataset()
+        context["form"] = form
+    return render(request, 'dataset_create.html',
+                  context)
+
+
+def select_biomarkers(request):
```

Comment:
it seems that this is a continuation of the `create` view, and wouldn't function as a separate view? Perhaps it should be part of the `create` view then? Perhaps, I'm not sure how you do this sort of dynamic form in django....